### ISSUE TITLE
Fedora 26 ember crash on startup "Could not load gui"

### ISSUE THREAD

**Issue by albiorix** [worldforge/ember#24]
Hi.  I've installed ember, cyphesis and dependencies from the packages supplied by the Fedora team.
The configuration screen comes up, I leave everything as-is, the program starts loading stuff and then fails.

Here are the messages -
Starting Ember....
Registering: list_commands
Registering: quit
Registering: toggle_erispolling
Starting Ember version 0.7.2
Initializing Ember services
Registering: loadscript
Registering: connect
Registering: meta_refresh
Registering: meta_abort
Metaserver Service created
Registering: set_value
Registering: get_value
Setting home directory to /home/neilt/.ember
libGL error: failed to create drawable
X Error of failed request:  0
  Major opcode of failed request:  155 (GLX)
  Minor opcode of failed request:  26 (X_GLXMakeContextCurrent)
  Serial number of failed request:  29
  Current serial number in output stream:  29
AL lib: (EE) alc_cleanup: 1 device not closed
[neilt@wol worldforge]$ less ~/.ember/ember.log 
[neilt@wol worldforge]$ > ~/.ember/ember.log;ember
Starting Ember....
Registering: list_commands
Registering: quit
Registering: toggle_erispolling
Starting Ember version 0.7.2
Initializing Ember services
Registering: loadscript
Registering: connect
Registering: meta_refresh
Registering: meta_abort
Metaserver Service created
Registering: set_value
Registering: get_value
Setting home directory to /home/neilt/.ember

I've looked in the ember.log file, and it fails immediately it tries to start CEGUI for the second time - 
.
.
.
[19:56:55] INFO Finished loading 214 modeldefinitions.
[19:56:55] INFO Registering: set_level
[19:56:55] INFO The material '/base/simple' has 4 supported techniques out of 4
[19:56:55] INFO The material '/base/simple' has best supported scheme High. Was looking for Experimental
[19:56:55] INFO The material '/base/simple' has 4 supported techniques out of 4
[19:56:55] INFO The material '/base/simple' supported with scheme High
[19:56:55] INFO The material '/base/normalmap' has 8 supported techniques out of 8
[19:56:55] INFO The material '/base/normalmap' supported with scheme High
[19:56:55] INFO The material '/base/normalmap/specular' has 8 supported techniques out of 8
[19:56:55] INFO The material '/base/normalmap/specular' supported with scheme High
[19:56:55] INFO Best graphics level is High
[19:56:55] INFO Using graphics level Medium
[19:56:55] INFO Registering: toggle_inputmode
[19:56:55] INFO Registering: reloadgui
[19:56:55] INFO Registering: toggle_gui
[19:56:55] INFO Binding key tab to command /console_focus for state gui.
[19:56:55] INFO Setting CEGUI logging level to 'standard'.
[19:56:55] INFO Starting CEGUI
[19:56:55] FAILURE (CEGUI) CEGUI::ScriptException in function 'static void CEGUI::LuaFunctor::pushNamedFunction(lua_
State*, const CEGUI::String&)' (/builddir/build/BUILD/cegui-0.8.7/cegui/src/ScriptModules/Lua/Functor.cpp:222) : Una
ble to get the Lua event handler: 'debug.traceback' as first part is not a table
[19:56:55] FAILURE GUIManager - error when creating gui.
        Exception: CEGUI::ScriptException in function 'static void CEGUI::LuaFunctor::pushNamedFunction(lua_State*, const CEGUI::String&)' (/builddir/build/BUILD/cegui-0.8.7/cegui/src/ScriptModules/Lua/Functor.cpp:222) : Unable to get the Lua event handler: 'debug.traceback' as first part is not a table
[19:56:55] INFO Deregistering: toggle_gui
[19:56:55] INFO Deregistering: reloadgui
[19:56:55] INFO Deregistering: toggle_inputmode
[19:56:55] CRITICAL Error during startup.
        Exception: Could not load gui, aborting. Make sure that all media got downloaded and installed correctly.
[19:56:55] INFO Service 'Server Service' stopped.
[19:56:55] INFO (Ogre) Unregistering ResourceManager for type SoundDefinition
[19:56:55] INFO (Ogre) Unregistering ResourceManager for type EntityRecipe
[19:56:55] INFO (Ogre) Unregistering ResourceManager for type TerrainLayerDefinition
[19:56:55] INFO (Ogre) Unregistering ResourceManager for type EntityMappingDefinition
[19:56:55] INFO (Ogre) Unregistering ResourceManager for type ModelDefinition
.
.
.

I have attached the entire ember.log -
[ember.log](https://github.com/worldforge/ember/files/1451109/ember.log)

**Comment by Anth0rx** [comment]
Before CEGUI fails there are a couple failures on Ogre, too:
```
[19:56:47] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib/OGRE/RenderSystem_GL.so.  System Error: /usr/lib/OGRE/RenderSystem_GL.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:47] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib/OGRE/RenderSystem_GL_d.so.  System Error: /usr/lib/OGRE/RenderSystem_GL_d.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:47] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib/OGRE/Plugin_ParticleFX.so.  System Error: /usr/lib/OGRE/Plugin_ParticleFX.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:47] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib/OGRE/Plugin_ParticleFX_d.so.  System Error: /usr/lib/OGRE/Plugin_ParticleFX_d.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:47] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib/OGRE/Plugin_CgProgramManager.so.  System Error: /usr/lib/OGRE/Plugin_CgProgramManager.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:47] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib/OGRE/Plugin_CgProgramManager_d.so.  System Error: /usr/lib/OGRE/Plugin_CgProgramManager_d.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:47] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib64/OGRE/Plugin_CgProgramManager.so.  System Error: /usr/lib64/OGRE/Plugin_CgProgramManager.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:47] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib64/OGRE/Plugin_CgProgramManager_d.so.  System Error: /usr/lib64/OGRE/Plugin_CgProgramManager_d.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:47] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/local/lib/OGRE/Plugin_CgProgramManager.so.  System Error: /usr/local/lib/OGRE/Plugin_CgProgramManager.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:47] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/local/lib/OGRE/Plugin_CgProgramManager_d.so.  System Error: /usr/local/lib/OGRE/Plugin_CgProgramManager_d.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:47] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib/OGRE/Plugin_CgProgramManager.so.  System Error: /usr/lib/OGRE/Plugin_CgProgramManager.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:47] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib/OGRE/Plugin_CgProgramManager_d.so.  System Error: /usr/lib/OGRE/Plugin_CgProgramManager_d.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:47] FAILURE Failed to load the plugin 'Plugin_CgProgramManager'!
[19:56:47] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib/OGRE/Plugin_OctreeSceneManager.so.  System Error: /usr/lib/OGRE/Plugin_OctreeSceneManager.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:47] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib/OGRE/Plugin_OctreeSceneManager_d.so.  System Error: /usr/lib/OGRE/Plugin_OctreeSceneManager_d.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:51] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib/OGRE/RenderSystem_GL.so.  System Error: /usr/lib/OGRE/RenderSystem_GL.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:51] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib/OGRE/RenderSystem_GL_d.so.  System Error: /usr/lib/OGRE/RenderSystem_GL_d.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:51] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib/OGRE/Plugin_ParticleFX.so.  System Error: /usr/lib/OGRE/Plugin_ParticleFX.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:51] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib/OGRE/Plugin_ParticleFX_d.so.  System Error: /usr/lib/OGRE/Plugin_ParticleFX_d.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:51] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib/OGRE/Plugin_CgProgramManager.so.  System Error: /usr/lib/OGRE/Plugin_CgProgramManager.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:51] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib/OGRE/Plugin_CgProgramManager_d.so.  System Error: /usr/lib/OGRE/Plugin_CgProgramManager_d.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:51] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib64/OGRE/Plugin_CgProgramManager.so.  System Error: /usr/lib64/OGRE/Plugin_CgProgramManager.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:51] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib64/OGRE/Plugin_CgProgramManager_d.so.  System Error: /usr/lib64/OGRE/Plugin_CgProgramManager_d.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:51] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/local/lib/OGRE/Plugin_CgProgramManager.so.  System Error: /usr/local/lib/OGRE/Plugin_CgProgramManager.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:51] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/local/lib/OGRE/Plugin_CgProgramManager_d.so.  System Error: /usr/local/lib/OGRE/Plugin_CgProgramManager_d.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:51] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib/OGRE/Plugin_CgProgramManager.so.  System Error: /usr/lib/OGRE/Plugin_CgProgramManager.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:51] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib/OGRE/Plugin_CgProgramManager_d.so.  System Error: /usr/lib/OGRE/Plugin_CgProgramManager_d.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:51] FAILURE Failed to load the plugin 'Plugin_CgProgramManager'!
[19:56:51] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib/OGRE/Plugin_OctreeSceneManager.so.  System Error: /usr/lib/OGRE/Plugin_OctreeSceneManager.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:51] FAILURE (Ogre) OGRE EXCEPTION(7:InternalErrorException): Could not load dynamic library /usr/lib/OGRE/Plugin_OctreeSceneManager_d.so.  System Error: /usr/lib/OGRE/Plugin_OctreeSceneManager_d.so: cannot open shared object file: No such file or directory in DynLib::load at /builddir/build/BUILD/ogre-1.9.0/OgreMain/src/OgreDynLib.cpp (line 111)
[19:56:55] FAILURE (CEGUI) CEGUI::ScriptException in function 'static void CEGUI::LuaFunctor::pushNamedFunction(lua_State*, const CEGUI::String&)' (/builddir/build/BUILD/cegui-0.8.7/cegui/src/ScriptModules/Lua/Functor.cpp:222) : Unable to get the Lua event handler: 'debug.traceback' as first part is not a table
[19:56:55] FAILURE GUIManager - error when creating gui.
```
So there seems to be a problem with OgreDynLib to load the respective shared libraries:
1. RenderSystem_GL.so
2. RenderSystem_GL_d.so
3. Plugin_ParticleFX.so.
4. Plugin_ParticleFX_d.so
5. Plugin_CgProgramManager.so
6. Plugin_CgProgramManager_d.so
7. Plugin_OctreeSceneManager.so
8. Plugin_OctreeSceneManager_d.so

Do you have the development headers for Ogre installed? You should also check `/usr/lib`, `/usr/lib64`, `/usr/local/lib` and `/usr/local/lib64` for the respective files.

**Comment by albiorix** [comment]
OK...I've looked carefully through this stuff again, the system tries to load its plugins from various places (/usr/lib, /usr/local/lib, /usr/lib64, /usr/local/lib64, etc.  It also varies the name from <libname>.so to <libname>_d.so until it either succeeds or runs out of options.  The only one it has run out of options on is CgProgramManager, which makes sense because that seems to be some kind of NVidia thing and I don't have any NVidia hardware.  I've dyked it out of the config file, and it no longer tries to load it.

The only real error now that I can see is - 
```
[20:50:04] INFO Setting CEGUI logging level to 'standard'.
[20:50:04] INFO Starting CEGUI
[20:50:04] FAILURE (CEGUI) CEGUI::ScriptException in function 'static void CEGUI::LuaFunctor::pushNamedFunction(lua_St
ate*, const CEGUI::String&)' (/builddir/build/BUILD/cegui-0.8.7/cegui/src/ScriptModules/Lua/Functor.cpp:222) : Unable 
to get the Lua event handler: 'debug.traceback' as first part is not a table
[20:50:04] FAILURE GUIManager - error when creating gui.
        Exception: CEGUI::ScriptException in function 'static void CEGUI::LuaFunctor::pushNamedFunction(lua_State*, const CEGUI::String&)' (/builddir/build/BUILD/cegui-0.8.7/cegui/src/ScriptModules/Lua/Functor.cpp:222) : Unable to get the Lua event handler: 'debug.traceback' as first part is not a table
[20:50:04] INFO Deregistering: toggle_gui
[20:50:04] INFO Deregistering: reloadgui
[20:50:04] INFO Deregistering: toggle_inputmode
[20:50:04] CRITICAL Error during startup.
        Exception: Could not load gui, aborting. Make sure that all media got downloaded and installed correctly.
[20:50:04] INFO Service 'Server Service' stopped.
[20:50:04] INFO (Ogre) Unregistering ResourceManager for type SoundDefinition
[20:50:04] INFO (Ogre) Unregistering ResourceManager for type EntityRecipe
[20:50:04] INFO (Ogre) Unregistering ResourceManager for type TerrainLayerDefinition
[20:50:04] INFO (Ogre) Unregistering ResourceManager for type EntityMappingDefinition
[20:50:04] INFO (Ogre) Unregistering ResourceManager for type ModelDefinition
```

**Comment by Anth0rx** [comment]
Okay, so there seems to be a Lua problem in `/builddir/build/BUILD/cegui-0.8.7/cegui/src/ScriptModules/Lua/Functor.cpp:222`. Do you have the proper lua packages installed?

**Comment by erikogenvik** [comment]
The error seems to be that the Lua module "debug" isn't available in Fedora by default.
I'll patch the current source to check for it, but in the meantime you can disable that functionality by adding the following to your ```~/.config/ember/ember.conf``` file.
```
[lua]
debug = false
```

**Comment by erikogenvik** [comment]
This issue is now also fixed in commit ca699b9652009a9bb3b4be226529f7eb16e67f4d
